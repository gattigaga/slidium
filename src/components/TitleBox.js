import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const InputTitle = styled.input`
  width: ${props => props.value.length * 9}px;
  height: 24px;
  background: white;
  padding: 4px;
  font-family: Roboto;
  border: 1px solid white;
  font-size: 18px;
  outline-width: 0px;

  &:hover,
  &:focus {
    border: 1px solid #ddd;
  }
`;

const TitleBox = ({ value, onChange }) => (
  <InputTitle
    type="text"
    value={value}
    onChange={onChange}
    onClick={e => e.target.select()}
  />
);

TitleBox.propTypes = {
  value: PropTypes.node,
  onChange: PropTypes.func
};

TitleBox.defaultProps = {
  value: "Untitled"
};

export default TitleBox;
