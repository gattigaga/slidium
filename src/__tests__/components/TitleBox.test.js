import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import TitleBox from "components/TitleBox";

let wrapper, onChange;

beforeEach(() => {
  onChange = jest.fn();
  wrapper = shallow(<TitleBox value="My Slides" onChange={onChange} />);
});

describe("TitleBox", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onChange'", () => {
    const event = {
      target: {
        value: "My Slides"
      }
    };

    expect(onChange.mock.calls.length).toEqual(0);
    wrapper.simulate("change", event);
    expect(onChange.mock.calls.length).toEqual(1);
  });
});
