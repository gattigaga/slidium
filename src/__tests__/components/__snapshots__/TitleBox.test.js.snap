// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBox renders correctly 1`] = `
<styled.input
  onChange={[Function]}
  onClick={[Function]}
  type="text"
  value="My Slides"
/>
`;
